src/main/java/uk/ac/ebi/eva/pipeline/io/writers/VariantStatsWriter.java

```diff
@@ -0,0 +1,46 @@
+package uk.ac.ebi.eva.pipeline.io.writers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.ItemWriter;
+import org.springframework.data.mongodb.core.BulkOperations;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.query.Update;
+import uk.ac.ebi.eva.commons.models.mongo.entity.VariantDocument;
+import uk.ac.ebi.eva.pipeline.parameters.DatabaseParameters;
+
+import java.util.List;
+
+public class VariantStatsWriter implements ItemWriter<VariantDocument> {
+    private static final Logger logger = LoggerFactory.getLogger(VariantStatsWriter.class);
+    private DatabaseParameters databaseParameters;
+    private MongoTemplate mongoTemplate;
+
+    public VariantStatsWriter(DatabaseParameters databaseParameters, MongoTemplate mongoTemplate) {
+        this.databaseParameters = databaseParameters;
+        this.mongoTemplate = mongoTemplate;
+    }
+
+    @Override
+    public void write(List<? extends VariantDocument> variants) {
+        BulkOperations bulkOperations = mongoTemplate.bulkOps(BulkOperations.BulkMode.UNORDERED, VariantDocument.class,
+                databaseParameters.getCollectionVariantsName());
+        for (VariantDocument variant : variants) {
+            if (variant.getVariantStatsMongo() == null || variant.getVariantStatsMongo().isEmpty()) {
+                continue;
+            }
+            Query query = new Query(Criteria.where("_id").is(variant.getId()));
+            Update update = new Update();
+            update.set("st", variant.getVariantStatsMongo());
```

Comment:
Potential optimisation: Compare the stats present before with the result of calculation and only update if the two are different.
Maybe a in a different ticket.

Reply:
In light of our discussion this morning, the approach we take will depend on how we decide to handle objects in the files array that share the same sid and fid. Specifically, whether we plan to merge these objects or keep them separate.

If we decide to keep them separate, it will become challenging to accurately map which stats belong to which files object, especially since multiple objects might have the same sid and fid